File: 백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.js
const N = +require('fs').readFileSync(0, 'utf-8').toString();
let K = 0;
let record = '';

move(1, 3, 2, N);
console.log(`${K}\n${record}`);

function move(start, end, other, depth) {
    if (depth === 0) return;
    
    move(start, other, end, depth - 1);
    K++;
    record += `${start} ${end}\n`;
    move(other, end, start, depth - 1);
}

File: 백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.py
N = int(input())
def move(s, e, r, d):
    if d == 1:
        print(f'{s} {e}')
        return
    
    move(s, r, e, d-1)
    print(f'{s} {e}')
    move(r, e, s, d-1)
    
print(2 ** N - 1)
move(1, 3, 2, N)